myapp/app/controllers/users_controller.rb

```diff
@@ -0,0 +1,18 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    user_data = params.require(:user).permit(:username, :password, :password_confirmation)
+
+    @user = User.new(user_data)
+
+    if @user.save 
+      log_in(@user)
+      redirect_to tasks_path, notice: 'Successfully signed up!' 
+    else 
+      render :new
+    end
+  end
```

Comment:
Tips: Sign up is handled by SessionController or AuthController generally. 
so that it does'nt mix with user CRUD functionality in UserController

Reply:
Fixed it in 9fb5d1e.
I can imagine resolving conflicts would be quite hard...

Reply:
I can imagine resolving conflicts would be quite hard...
> Then its good to be master of it, Its a part of our learning though :)